src/inmanta/deploy/work.py

```diff
@@ -406,42 +459,63 @@ def extend_blocked_on(resource: ResourceIdStr, new_blockers: set[ResourceIdStr])
+                    new_priority, new_tiebreaker, new_reason = (
+                        discarded.priority,
+                        discarded.priority_tiebreaker,
+                        discarded.reason,
+                    )
+                else:
+                    # Get priority from running task. This should always return something, but default to this request's
+                    # priority just in case
+                    new_priority = self.agent_queues.in_progress.get(task, priority)
+                    new_tiebreaker = None
+                    new_reason = "rescheduling because a dependency was added to this resource while it was deploying"
+
                 queued.remove(resource)
                 self._waiting[resource] = BlockedDeploy(
-                    task=PrioritizedTask(
-                        task=task,
-                        priority=min(task_priority, priority) if task_priority is not None else priority,
-                        reason=reason,
-                    ),
+                    task=task,
                     # task was previously ready to execute => assume no other blockers than this one
                     blocked_on=new_blockers,
+                    priority=new_priority,
+                    priority_tiebreaker=new_tiebreaker,
+                    reason=new_reason,
                 )
             else:
                 self._waiting[resource].blocked_on.update(new_blockers)
                 maybe_runnable.discard(resource)
 
         # ensure desired resource deploys are scheduled
         for resource in resources:
-            prioritized_task = PrioritizedTask(task=tasks.Deploy(resource=resource), priority=priority, reason=reason)
+            task = tasks.Deploy(resource=resource)
             if is_scheduled(resource):
                 # Deploy is already scheduled / running. Check to see if this task has a higher priority than the one already
                 # scheduled. If it has, update the priority. If any of its dependencies are to be
                 # scheduled as well, they will follow the provides relation to ensure this deploy waits its turn.
-                if prioritized_task.task in self.agent_queues:
+                if task in self.agent_queues:
                     # simply add it again, the queue will make sure only the highest priority is kept
-                    self.agent_queues.queue_put_nowait(prioritized_task)
-                if resource in self._waiting:
-                    if self._waiting[resource].task.priority > priority:
-                        self._waiting[resource].task = prioritized_task
+                    self.agent_queues.queue_put_nowait(task, priority=priority, reason=reason)
+                if resource in self._waiting and priority < self._waiting[resource].priority:
+                    self._waiting[resource].priority = priority
+                    self._waiting[resource].priority_tiebreaker = None
```

Comment:
I would expect the tie breaker to remain when bumping priority?

Reply:
We're already moving to a more urgent bucket, so we're moving forward from where we're scheduled. I wouldn't expect it to jump the queue of that bucket, considering that other items in the bucket were requested first *for that priority*.

Does that reasoning change your mind, or should we take this to Slack?

Reply:
Discussed further in person. Conclusion: current resetting of the tiebreaker is good, but we should assign it when a task gets scheduled rather than when it gets queued. Reworked in 507ea3126.